src/HealthChecks.AzureServiceBus/AzureServiceBusSubscriptionHealthCheck.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+
+namespace HealthChecks.AzureServiceBus
+{
+    public class AzureServiceBusSubscriptionHealthCheck : IHealthCheck
+    {
+        private static readonly ConcurrentDictionary<string, ManagementClient> ManagementClientConnections =
+            new ConcurrentDictionary<string, ManagementClient>();
+        private readonly string _connectionString;
+        private readonly string _topicName;
+        private readonly string _subscriptionName;
+
+
+        public AzureServiceBusSubscriptionHealthCheck(string connectionString, string topicName, string subscriptionName)
+        {
+            if (string.IsNullOrEmpty(connectionString))
+            {
+                throw new ArgumentNullException(nameof(connectionString));
```

Comment:
If `connectionString` is empty then it should not be `ArgumentNullException`. The same for `topicName` and `subscriptionName`.

Reply:
What exception would you expect in that case?

Reply:
ArgumentException

Reply:
I don't have any preference but ArgumentNullException is used in other healthcheck classes

* https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.AzureServiceBus/AzureServiceBusQueueHealthCheck.cs#L25
* https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/master/src/HealthChecks.AzureServiceBus/AzureServiceBusTopicHealthCheck.cs#L27

If I change this to ArgumentException, then we would have to update everywhere else, I'm afraid

Reply:
If `ArgumentNullException` is used for arguments other than `null` then it is obviously wrong.

Reply:
I am happy to change them all if other maintainers are happy